src/linter/dotLibrary/DotLibraryLinter.ts

```diff
@@ -0,0 +1,99 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import {deprecatedLibraries} from "../../utils/deprecations.js";
+import {SaxEventType, Tag as SaxTag} from "sax-wasm";
+import {parseXML} from "../../utils/xmlParser.js";
+import {ReadStream} from "node:fs";
+import {RULES, MESSAGES, formatMessage} from "../linterReporting.js";
+
+export default class DotLibraryLinter {
+	#contentStream;
+	#resourcePath;
+	#context: LinterContext;
+
+	constructor(resourcePath: string, contentStream: ReadStream, context: LinterContext) {
+		this.#contentStream = contentStream;
+		this.#resourcePath = resourcePath;
+		this.#context = context;
+	}
+
+	async lint() {
+		try {
+			const dotLibraryDependencyTags = await this.#parseDotLibrary(this.#contentStream);
+			this.#analyzeDeprecatedLibs(dotLibraryDependencyTags);
+		} catch (err) {
+			const message = err instanceof Error ? err.message : String(err);
+			this.#context.addLintingMessage(this.#resourcePath, {
+				severity: LintMessageSeverity.Error,
+				message,
+				ruleId: RULES["ui5-linter-parsing-error"],
+				fatal: true,
+			});
+		}
+	}
+
+	async #parseDotLibrary(contentStream: ReadStream): Promise<SaxTag[]> {
+		const libs = new Set();
+		const tagsStack: string[] = [];
+		const libNamePath = ["library", "dependencies", "dependency"];
+		await parseXML(contentStream, (event, tag) => {
+			if (!(tag instanceof SaxTag)) {
+				return;
+			}
+
+			if (event === SaxEventType.OpenTag && !tag.selfClosing) {
+				tagsStack.push(tag.value);
+			} else if (event === SaxEventType.CloseTag && !tag.selfClosing) {
+				tagsStack.pop();
+			}
+
+			if (event === SaxEventType.CloseTag &&
+				tag.value === "libraryName") {
+				const path = tagsStack.slice(-1 * libNamePath.length);
+				const isMatchingPath = libNamePath.every((lib, index) => lib === path[index]);
+
+				if (isMatchingPath) {
+					libs.add(tag);
+				}
+			}
+		});
+
+		return Array.from(libs) as SaxTag[];
+	}
+
+	#analyzeDeprecatedLibs(libs: SaxTag[]) {
+		// Check for deprecated libraries
+		libs.forEach((lib) => {
+			const {line, character: column} = lib.openStart;
+
+			// Check for missing textNodes or for empty textNodes withing the
+			// libraryName tag
+			if (!lib.textNodes[0] || lib.textNodes[0].value.trim() === "") {
+				this.#context.addLintingMessage(this.#resourcePath, {
+					ruleId: RULES["ui5-linter-empty-library-name"],
+					severity: LintMessageSeverity.Warning,
+					fatal: undefined,
+					line: line + 1,
+					column: column + 1,
+					message: formatMessage(MESSAGES.SHORT__EMPTY_LIBRARY_NAME),
+					messageDetails: formatMessage(MESSAGES.DETAILS__EMPTY_LIBRARY_NAME),
+				});
+
+				return;
+			}
+
+			const libName = lib.textNodes[0].value.trim();
+
+			if (deprecatedLibraries.includes(libName)) {
+				this.#context.addLintingMessage(this.#resourcePath, {
+					ruleId: RULES["ui5-linter-no-deprecated-api"],
```

Comment:
That is not the ID that corresponds to the message, right? To me it seems the `addLintingMessage` API should become easier to use. Right now it's error prone to manually format the message and use the correct corresponding ruleId.

Reply:
Thanks! Fixed the ruleId

Maybe, we can discuss how to handle this as I see many ways to do it and it'd always require some level of attention and detail

Reply:
Yes, it's not completely straight-forward. I'd like to hear your ideas.

Reply:
At first sight I was thinking about having some kind of currying, so that context's `addLinteMessage` would come in side the specific linting module preconfigured. However there are cases where we detect multiple things and that might be the challenge here.

Reply:
I've created https://github.com/SAP/ui5-linter/pull/251 with some rough idea I had. But we should probably also think about introducing a base class for the reporters so that we can streamline the signatures.

I also had the idea to only collect the information necessary to create the messages during linting, and only produce the text and details when creating the report. This could reduce the memory usage during linting, which is already very high. The enum would help with that, as it is technically just a number in JS.
But this can easily be done in a follow-up refactoring, as it most probably won't affect the `addMessage` signature.